handlers/people-handler/people_handler.py

```diff
@@ -0,0 +1,484 @@
+            rendering += sentence
+
+    print("final output is: ")
+    print(rendering)
+    print("\n")
+    return rendering
+
+
+@app.route("/handler", methods=["POST"])
+def handle():
+    logging.debug("Received request")
+    contents = request.get_json()
+
+    # Check preprocessor data
+    preprocessors = contents['preprocessors']
+    possible_people = 0
+
+    # No Object Detector found
+    prep = "ca.mcgill.a11y.image.preprocessor.objectDetection"
+    if prep not in preprocessors:
+        logging.debug("No Object Detector found")
+        logging.debug("Sending response")
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        return response
+
+    if ("ca.mcgill.a11y.image.preprocessor.graphicTagger" in preprocessors):
+        if (preprocessors["ca.mcgill.a11y.image.preprocessor.graphicTagger"]
+                ["category"] == "people"):
+            objects = preprocessors[prep]["objects"]
+            possible_people += 1
+
+    # checks if emotion detection detects face
+    possible_people += c.custom_check(preprocessors)
+    objects = preprocessors[prep]["objects"]
+    res = preprocessors["ca.mcgill.a11y.image.preprocessor.caption"]["caption"]
+    if (possible_people <= 1):
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": [
+                {
+                    "type_id": "ca.mcgill.a11y.image.renderer.Text",
+                    "description": "The text found in a graphic.",
+                    "data": {
+                        "text": "Cannot be rendered"
+                    }
+                }
+            ]
+        }
+        return response
+    else:
+        sort = "ca.mcgill.a11y.image.preprocessor.sorting"
+        objects, left2right = f.remove_low_confidence(
+            objects, preprocessors[sort]["leftToRight"])
+        objects_sorted = sorted(objects, key=lambda d: d['area'], reverse=True)
+        change = per_change(objects_sorted)
+        if ("ca.mcgill.a11y.image.preprocessor.emotion" in preprocessors):
+            # print(preprocessors['ca.mcgill.a11y.image.preprocessor.position']["data"])
+            preprocessors = f.get_original_format(preprocessors)
+            mf, ef, oei, objects, just_person_count, cf = f.format_json(
+                objects_sorted, change, preprocessors)
+            multiple_flag = mf
+            emotion_flag = ef
+            cloth_flag = cf
+            object_emotion_inanimate = oei
+            rendering = get_rendering(
+                multiple_flag,
+                emotion_flag,
+                object_emotion_inanimate,
+                preprocessors,
+                objects,
+                just_person_count,
+                cloth_flag,
+                contents,
+                res)
+            logging.critical(rendering)
```

Comment:
CRITICAL??? This is **absolutely** not right.

Reply:
@jeffbl `logging.critical` seems to work. What exactly is wrong with this approach?

<img width="1427" alt="Screenshot 2023-10-10 at 10 17 18 PM" src="https://github.com/Shared-Reality-Lab/IMAGE-server/assets/27781159/eaf3874c-75ec-4163-a6a9-f37df620b712">

Reply:
[Python log levels documented here](https://docs.python.org/3/library/logging.html#logging-levels). A critical log message is typically reserved for something that crashes the system or results in user data loss. This is a completely normal, in fact successful, condition. I'd say this is log level info.